File: diff_scripts/diff.py
import sys

def match_pct(actual, expected):
    with open(actual, mode='rb') as file:
        actualContent = file.read()
    with open(expected, mode='rb') as file:
        expectedContent = file.read()

    diff_count = 0.0
    for i in range(0, min(len(actualContent), len(expectedContent))):
        if actualContent[i] != expectedContent[i]:
            diff_count += 1
    
    size_diff = abs(len(actualContent) - len(expectedContent))
    
    diff_count += size_diff
    diff_pct = diff_count/len(expectedContent)
    if diff_pct > 1.0:
        print('diff_pct > 1.0, treating as 1.0')
        diff_pct = 1.0
        
    match_pct = 1.0 - diff_pct
    return abs(round(match_pct, 6))

if __name__ == "__main__":
    actual = sys.argv[1]
    expected = sys.argv[2] 
    print('Input: {}\nExpected: {}\nMatch: {}'.format(actual, expected, match_pct(actual, expected)))

File: diff_scripts/diff.ts
function matchPct(actual: string, expected: string): string {
    const actualContent = Deno.readFileSync(actual);
    const expectedContent = Deno.readFileSync(expected);

    let diffCount = 0;
    for(let i = 0; i < Math.min(actualContent.length, expectedContent.length); i++) {
        if(actualContent[i] !== expectedContent[i]) {
            diffCount++;
        }
    }

    const sizeDiff = Math.abs(actualContent.length - expectedContent.length);
    diffCount += sizeDiff;
    let diffPct = diffCount/expectedContent.length;
    if(diffPct > 1.0) {
        console.log('diffPct > 1.0, treating as 1.0');
        diffPct = 1.0;
    }

    const matchPct = 1.0 - diffPct;
    return Math.abs(matchPct).toFixed(6);
}

const actual = Deno.args[0];
const expected = Deno.args[1];
console.log(`Input: ${actual}
Expected: ${expected}
Match: ${matchPct(actual, expected)}`);